### PR TITLE
Subscribe all configuration keys when empty list is received.

### PR DESCRIPTION
**Pull request by CrazyHZM**
# Description

Subscribe all configuration keys when a request with an empty list is received.

## Issue reference

Fix https://github.com/dapr/dapr/issues/4178

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
@LaurenceLiZhixin please review. Thanks!

**Comment by CrazyHZM** [comment]
> @CrazyHZM please add tests to cover this case.

The unit test has been covered, do you mean e2e?

**Comment by daixiang0** [comment]
Try add e2e tests if possible.

**Comment by yaron2** [comment]
@CrazyHZM can you please resolve conflicts? also I think tests are failing..

**Comment by CrazyHZM** [comment]
> @CrazyHZM can you please resolve conflicts? also I think tests are failing..

@yaron2 Conflict has been fixed.

File: pkg/grpc/api.go [hunk 1]
```diff
@@ -1377,13 +1377,45 @@ func (a *api) SubscribeConfigurationAlpha1(request *runtimev1pb.SubscribeConfigu
...
+		for _, k := range subscribeKeys {
+			if _, ok := a.configurationSubscribe[fmt.Sprintf("%s||%s", request.StoreName, k)]; !ok {
+				unsubscribedKeys = append(unsubscribedKeys, k)
+			}
+		}
+	}
+
 	req := &configuration.SubscribeRequest{
-		Keys:     request.Keys,
+		Keys:     unsubscribedKeys,
```
Comment (artursouza) [inline pkg/grpc/api.go:1418]: Why is this called "unsubscribedKeys"? Is it being unsubscribed? This might just be a naming issue.

**Review by artursouza — CHANGES_REQUESTED** [review]

File: pkg/grpc/api.go (hunk 1 above)
Comment (CrazyHZM) [reply pkg/grpc/api.go]: @artursouza I have renamed the variable. Please review this.

**Review by artursouza — APPROVED** [review]

**Pull request merged**